### ISSUE TITLE
build failure/error using clang

### ISSUE THREAD

**Issue by ryzom-pipeline** [ryzom/ryzomcore#33]
**[Original report](https://bitbucket.org/ryzom/ryzomcore/issue/33) by John Smith (Bitbucket: [lbalbalba](https://bitbucket.org/lbalbalba), GitHub: [lbalbalba](https://github.com/lbalbalba)).**

----------------------------------------

The following compile failure/error using llvm/clang occurs:

```
#!shell

[ 51%] Building CXX object ryzom/client/src/CMakeFiles/ryzom_client.dir/cdb_synchronised.cpp.o
In file included from /usr/local/src/ryzomcore/code/ryzom/client/src/interface_v3/interface_manager.h:49:0,
                 from /usr/local/src/ryzomcore/code/ryzom/client/src/cdb_synchronised.cpp:34:
/usr/local/src/ryzomcore/code/ryzom/client/src/interface_v3/../string_manager_client.h:344:26: warning: multi-character character constant [-Wmultichar]
/usr/local/src/ryzomcore/code/ryzom/client/src/interface_v3/../string_manager_client.h:345:26: warning: multi-character character constant [-Wmultichar]
In file included from /usr/local/src/ryzomcore/code/ryzom/client/src/interface_v3/interface_manager.h:41:0,
                 from /usr/local/src/ryzomcore/code/ryzom/client/src/cdb_synchronised.cpp:34:
/usr/local/src/ryzomcore/code/ryzom/client/src/interface_v3/interface_config.h:94:22: error: field âExtraDatasâ has incomplete type
In file included from /usr/local/src/ryzomcore/code/ryzom/client/src/interface_v3/interface_manager.h:49:0,
                 from /usr/local/src/ryzomcore/code/ryzom/client/src/cdb_synchronised.cpp:34:
/usr/local/src/ryzomcore/code/ryzom/client/src/interface_v3/../string_manager_client.h:113:23: error: expected â;â at end of member declaration
/usr/local/src/ryzomcore/code/ryzom/client/src/interface_v3/../string_manager_client.h:113:58: error: âEGSPDâ has not been declared
/usr/local/src/ryzomcore/code/ryzom/client/src/interface_v3/../string_manager_client.h:113:106: error: expected â)â before âtypeâ
In file included from /usr/local/src/ryzomcore/code/ryzom/client/src/cdb_synchronised.cpp:34:0:
/usr/local/src/ryzomcore/code/ryzom/client/src/interface_v3/interface_manager.h:397:33: error: âCHARACTERISTICSâ has not been declared
/usr/local/src/ryzomcore/code/ryzom/client/src/interface_v3/interface_manager.h:397:67: error: expected â,â or â...â before âtypeâ
/usr/local/src/ryzomcore/code/ryzom/client/src/interface_v3/interface_manager.h:408:31: error: âCHARACTERISTICSâ has not been declared
/usr/local/src/ryzomcore/code/ryzom/client/src/interface_v3/interface_manager.h:408:65: error: expected â,â or â...â before âtypeâ
/usr/local/src/ryzomcore/code/ryzom/client/src/interface_v3/interface_manager.h:521:2: error: expected class-name before â{â token
/usr/local/src/ryzomcore/code/ryzom/client/src/interface_v3/interface_manager.h:636:30: error: âCHARACTERISTICSâ has not been declared
/usr/local/src/ryzomcore/code/ryzom/client/src/interface_v3/interface_manager.h: In member function âbool CInterfaceManager::isItemCaracRequirementMet(int)â:
/usr/local/src/ryzomcore/code/ryzom/client/src/interface_v3/interface_manager.h:400:7: error: âtypeâ was not declared in this scope
/usr/local/src/ryzomcore/code/ryzom/client/src/interface_v3/interface_manager.h:400:14: error: âCHARACTERISTICSâ has not been declared
/usr/local/src/ryzomcore/code/ryzom/client/src/interface_v3/interface_manager.h:401:11: error: âvalueâ was not declared in this scope
/usr/local/src/ryzomcore/code/ryzom/client/src/interface_v3/interface_manager.h:401:20: error: â_CurrentPlayerCharacâ was not declared in this scope
/usr/local/src/ryzomcore/code/ryzom/client/src/interface_v3/interface_manager.h: In member function âsint32 CInterfaceManager::getCurrentPlayerCarac(int)â:
/usr/local/src/ryzomcore/code/ryzom/client/src/interface_v3/interface_manager.h:410:7: error: âtypeâ was not declared in this scope
/usr/local/src/ryzomcore/code/ryzom/client/src/interface_v3/interface_manager.h:410:14: error: âCHARACTERISTICSâ has not been declared
/usr/local/src/ryzomcore/code/ryzom/client/src/interface_v3/interface_manager.h:411:11: error: â_CurrentPlayerCharacâ was not declared in this scope
/usr/local/src/ryzomcore/code/ryzom/client/src/interface_v3/interface_manager.h: In constructor âCInterfaceManager::CEmoteCmd::CEmoteCmd(const char*, const char*, const char*)â:
/usr/local/src/ryzomcore/code/ryzom/client/src/interface_v3/interface_manager.h:524:22: error: expected class-name before â(â token
/usr/local/src/ryzomcore/code/ryzom/client/src/interface_v3/interface_manager.h:524:22: error: expected â{â before â(â token
/usr/local/src/ryzomcore/code/ryzom/client/src/cdb_synchronised.cpp: In member function âvoid CCDBSynchronised::read(const string&)â:
/usr/local/src/ryzomcore/code/ryzom/client/src/cdb_synchronised.cpp:111:13: error: variable âstd::ifstream fâ has initializer but incomplete type
/usr/local/src/ryzomcore/code/ryzom/client/src/cdb_synchronised.cpp:111:31: error: incomplete type âstd::ios {aka std::basic_ios<char>}â used in nested name specifier
make[2]: *** [ryzom/client/src/CMakeFiles/ryzom_client.dir/cdb_synchronised.cpp.o] Error 1
make[1]: *** [ryzom/client/src/CMakeFiles/ryzom_client.dir/all] Error 2
make: *** [all] Error 2

```

**Comment by ryzom-pipeline** [comment]
**Original comment by Cédric Ochs (Bitbucket: [Cédric OCHS](https://bitbucket.org/Cédric OCHS), ).**

----------------------------------------

Please did you disable PCH ? Because it seams like :(

**Comment by ryzom-pipeline** [comment]
**Original comment by Cédric Ochs (Bitbucket: [Cédric OCHS](https://bitbucket.org/Cédric OCHS), ).**

----------------------------------------

fixes #33 build failure/error using clang
